pkg/k8s/client/cell.go

```diff
@@ -503,6 +514,37 @@ func NewStandaloneClientset(cfg Config) (Clientset, error) {
 	return clientset, err
 }
 
+type ClientBuilderFunc func(string) Clientset
+
+// NewClientBuilder returns a function that creates a new Clientset with the given
+// name as the user agent, or logs a fatal error if the Clientset cannot be
+// created.
+func NewClientBuilder(lc cell.Lifecycle, log logrus.FieldLogger, cfg Config) ClientBuilderFunc {
+	return func(name string) Clientset {
+		// Note: We are only making a shallow copy of the Config. This will need
+		// to be replaced with a DeepCopy method if any pointer types are
+		// introduced to Config.
+		newCfg := cfg
+		if name != "" {
+			newCfg.UserAgent = name
```

Comment:
nit: I would probably always add "default" user-name with version + specific controller name, similar to kube-controller-manager:
```
userAgent="kube-controller-manager/v1.27.3 (linux/amd64) kubernetes/25b4e43/leader-election"
kube-controller-manager/v1.27.3 (linux/amd64) kubernetes/25b4e43/system:serviceaccount:kube-system:resourcequota-controller"
```
You can check that with:
```
k logs -n kube-system kube-apiserver-kind-control-plane -f | grep cilium-agent
k logs -n kube-system kube-apiserver-kind-control-plane -f | grep kube-controller-manager
```
in our kind development.

Reply:
We're doing this already, but it's in `createConfig` here: https://github.com/cilium/cilium/blob/1856f8234825fc7115d1ad862fead09976accc70/pkg/k8s/client/cell.go#L307

Reply:
but only if UserAgent is empty. In this builder we are passing name that is assigned to UserAgent, so for example here:
https://github.com/cilium/cilium/pull/32353/files#diff-fd8b4296e348044f72c60ce118ad389fa5aa9633376bb1d4159c74005ea4e4afR79
UserAgent will be just `ciliumendpointslice-controller`, no?

Reply:
Hh my bad, I misunderstood your suggestion. I'll add this :+1:

Reply:
Restructured a bit to make the string passed to the ClientBuilderFunc be appended onto the full user agent. I dropped the user-agent flag, which I did not like much anyway